devel/t-dynamic/PERL-790-unacknowledged-write-sessions.t

```diff
@@ -0,0 +1,126 @@
+#
+#  Copyright 2015 MongoDB, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+use strict;
+use warnings;
+use Test::More 0.96;
+use Test::Fatal;
+use Test::Deep qw/!blessed/;
+use UUID::Tiny ':std'; # Use newer interface
+
+use utf8;
+use Tie::IxHash;
+
+use MongoDB;
+use MongoDB::Error;
+use MongoDB::_Types qw/ to_IxHash /;
+
+use lib "t/lib";
+use lib "devel/lib";
+
+use if $ENV{MONGOVERBOSE}, qw/Log::Any::Adapter Stderr/;
+
+use MongoDBTest::Orchestrator; 
+
+use MongoDBTest qw/
+    build_client
+    get_test_db
+    server_version
+    server_type
+    clear_testdbs
+    get_unique_collection
+/;
+
+# This test starts servers on localhost ports 27017, 27018 and 27019. We skip if
+# these aren't available.
+
+my $orc =
+MongoDBTest::Orchestrator->new(
+  config_file => "devel/config/replicaset-single-3.6.yml" );
+$orc->start;
+
+$ENV{MONGOD} = $orc->as_uri;
+
+print $ENV{MONGOD};
+
+my $conn           = build_client();
+my $testdb         = get_test_db($conn);
+my $server_version = server_version($conn);
+my $server_type    = server_type($conn);
+
+plan skip_all => "Requires MongoDB 3.6"
+    if $server_version < v3.6.0;
+
+subtest 'Session for ack writes' => sub {
+    use Test::Role::BSONDebug;
```

Comment:
use is like BEGIN { ... } so you can pull it (and the role application) out of the subtest block

Reply:
Ah yes, i'd just not got round to moving it all the way to the top heh. Moved!